packages/nuxt/src/app/composables/ssr.ts

```diff
@@ -1,13 +1,14 @@
 import type { H3Event } from 'h3'
-import { setResponseStatus as _setResponseStatus } from 'h3'
+import { setResponseStatus as _setResponseStatus, getRequestHeaders } from 'h3'
 import type { NuxtApp } from '../nuxt'
 import { useNuxtApp } from '../nuxt'
 
 export function useRequestHeaders<K extends string = string> (include: K[]): { [key in Lowercase<K>]?: string }
 export function useRequestHeaders (): Readonly<Record<string, string>>
 export function useRequestHeaders (include?: any[]) {
   if (import.meta.client) { return {} }
-  const headers = useNuxtApp().ssrContext?.event.node.req.headers ?? {}
+  const event = useNuxtApp().ssrContext?.event
```

Comment:
we can likely tree shake this by doing

```suggestion
  const event = import.meta.server && useNuxtApp().ssrContext?.event
```